source/Projectile.cpp

```diff
@@ -199,6 +198,43 @@ void Projectile::Move(vector<Visual> &visuals, vector<Projectile> &projectiles)
 			}
 		}
 	}
+	// Homing weapons that have lost their lock have a chance to get confused
+	// and turn in a random direction ("go haywire"). Each tracking method has
+	// a different haywire condition. Weapons with multiple tracking methods
+	// only go haywire if all of the tracking methods have gotten confused.
+	else if(target && homing)
+	{
+		bool infraredConfused = true;
+		bool opticalConfused = true;
+		bool radarConfused = true;
+
+		// Infrared: proportional to tracking quality
+		if (weapon->InfraredTracking())
+			infraredConfused = Random::Real() > weapon->InfraredTracking();
+
+		// Optical: proportional tracking quality
+		if (weapon->OpticalTracking())
+			opticalConfused = Random::Real() > weapon->OpticalTracking();
+
+		// Radar: If the target has no jamming, then proportional to tracking
+		// quality. If the target does have jamming, then it's proportional to
+		// tracking quality, the strength of target's jamming, and the distance
+		// to the target (jamming power attenuates with distance).
+		if (weapon->RadarTracking())
+		{
+			if (target->Attributes().Get("radar jamming") == 0)
+				radarConfused = Random::Real() > weapon->RadarTracking();
+			else
+			{
+				// Closeness to target; 0 when just fired, 1 when impacting
+				double closenessTotarget = 1 + weapon->RadarTracking() - (1 / (weapon->Range() / position.Distance(target->Position())));
+
+				radarConfused = Random::Real() < closenessTotarget * ( 1 - (weapon->RadarTracking() / target->Attributes().Get("radar jamming")));
+			}
+		}
+		if (infraredConfused && opticalConfused && radarConfused)
+			turn = Random::Real() - min(.5, turn);
```

Comment:
I'm not sure what was meant to be set as value here, but turn is used as input for angle, so I would expect the calculation here to use turn-values in the range of -180 to 180 or in the range 0 to 360 and I don't recognize those ranges.

Reply:
This gently modifies the existing missile's existing turn value to cause it to veer off of its current course without making it look like it's totally spazzing out.